app/models/ArrivalUpdate.scala

```diff
@@ -17,11 +17,44 @@
 package models
 
 import cats._
-import play.api.libs.json.JsObject
 import play.api.libs.json.Json
 import play.api.libs.json.Writes
 
-case class MessageStatusUpdate(messageId: MessageId, messageStatus: MessageStatus)
+sealed trait ArrivalUpdate
+
+object ArrivalUpdate {
+
+  implicit val arrivalUpdateSemigroup: Semigroup[ArrivalUpdate] = {
+    case (ArrivalStatusUpdate(_), x @ ArrivalStatusUpdate(_))        => x
+    case (a @ ArrivalStatusUpdate(_), m @ MessageStatusUpdate(_, _)) => CompoundStatusUpdate(a, m)
+    case (ArrivalStatusUpdate(_), c @ CompoundStatusUpdate(_, _))    => c
+    case (ArrivalStatusUpdate(_), p @ ArrivalPutUpdate(_, _))        => p
+
```

Comment:
Are there any plans on using the params? As I feel the Object(_, _) format makes this difficult to read compared to something like:

(Case _: ArrivalStatusUpdate, _: ArrivalStatusUpdate)